client/layout/sidebar/button.jsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { isExternal } from 'lib/url';
+import { preload } from 'sections-preload';
+
+export default React.createClass( {
+	displayName: 'SidebarButton',
+
+	propTypes: {
+		href: React.PropTypes.string,
+		onClick: React.PropTypes.func,
+		preloadSectionName: React.PropTypes.string,
+		children: React.PropTypes.node
+	},
+
+	_preloaded: false,
+
+	preload() {
+		if ( ! this._preloaded && this.props.preloadSectionName ) {
+			this._preloaded = true;
+			preload( this.props.preloadSectionName );
```

Comment:
Probably fine for another PR, but I feel we should just enhance `sections-preload` to track sections already preloaded, rather than at every instance where we're invoking the preload.